core/src/main/scala/pureconfig/backend/ConfigFactoryWrapper.scala

```diff
@@ -34,7 +34,13 @@ object ConfigFactoryWrapper {
   private def unsafeToEither[A](f: => A, path: Option[Path] = None): Either[ConfigReaderFailures, A] = {
     try Right(f) catch {
       case e: ConfigException.IO if path.nonEmpty => fail(CannotReadFile(path.get, Option(e.getCause)))
-      case e: ConfigException.Parse => fail(CannotParse(e.getLocalizedMessage, ConfigValueLocation(e.origin())))
+      case e: ConfigException.Parse =>
+        val msg = (if (e.origin != null)
+          // Removing the error origin from the exception message since origin is stored and used separately:
+          e.getMessage.stripPrefix(s"${e.origin.description}: ")
+        else
+          e.getMessage).stripSuffix(".")
```

Comment:
I'll link the source code of [`ConfigException`](https://github.com/typesafehub/config/blob/master/config/src/main/java/com/typesafe/config/ConfigException.java#L21) in case somebody else wants to check why we need do strip the prefix in this way. `ConfigException` takes in the constructor origin and [appends the description to the exception message](https://github.com/typesafehub/config/blob/master/config/src/main/java/com/typesafe/config/ConfigException.java#L23).